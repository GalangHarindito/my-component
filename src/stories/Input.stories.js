import { Input } from "../component/input/Input";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Input',
  component: Input,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    onChange: {
      description: 'Handle behaviour when input event',
      action: 'type',
    },
    placeholder: {
      description: 'Name for placeholder',
      action: 'string',
    },
    label: {
      description: 'Name for input label',
      action: 'string',
    },
    size: {
      description: 'Size for input',
      action: 'string',
    },
    error: {
      description: 'Error message if any',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
    label: 'Label Text',
    placeholder: 'Input Name',
    error: ''
  },
};

export const Medium = {
    args: {
      size: 'medium',
      label: 'Label Text',
      placeholder: 'Input Name',
      error: ''
    },
  };