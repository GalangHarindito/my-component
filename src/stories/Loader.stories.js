import { Loader } from "../component/spinner/loader";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
export default {
  title: 'Component/Loader',
  component: Loader,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes

  argTypes:{
    size: {
      description: 'Size for loader',
      action: 'string',
    }
  }
};

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args

export const Large = {
  args: {
    size: 'large',
  },
};

export const Medium = {
    args: {
      size: 'medium',
    },
  };

  export const Small = {
    args: {
      size: 'small',
    },
  };