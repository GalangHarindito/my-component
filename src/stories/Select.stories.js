import { SelectInput } from "../component/select/select";

export default {
    title: 'Component/Select',
    component: SelectInput,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
    tags: ['autodocs'],
    // More on argTypes: https://storybook.js.org/docs/api/argtypes
  
    argTypes:{
        onChange: {
            description: 'Handle behaviour when input event',
            action: 'type',
          },
          label: {
            description: 'Name for input label',
            action: 'string',
          },
          size: {
            description: 'Size for input',
            action: 'string',
          },
          error: {
            description: 'Error message if any',
            action: 'string',
          },
          required: {
            description: 'Required field',
            action: 'boolean',
          },
    }
  };
  
  // More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
  
  export const Select = {
    args: {
      size: 'large',
      label: 'Label Text',
      defaultValue: ' Input Value',
      error: '', 
      required: false,
      options: [
        { icon: "ðŸ¤£", value: "test1", name: "test1" },
        { icon: "ðŸ˜†", value: "test2", name: "test2" },
        { icon: "ðŸ¥°", value: "test3", name: "test3" },
      ]
    },
  };