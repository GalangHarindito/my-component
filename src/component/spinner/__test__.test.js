import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';
import * as stories from '../../stories/Loader.stories';// import all stories from the stories file

const { Large, Medium, Small } = composeStories(stories);

test('Make sure input UI does not change unexpectedly', () => {
    const { container } = render(<Large />)
    expect(container).toMatchSnapshot()
})

test('Make sure input UI does not change unexpectedly', () => {
    const { container } = render(<Medium />)
    expect(container).toMatchSnapshot()
})

test('Make sure input UI does not change unexpectedly', () => {
    const { container } = render(<Small />)
    expect(container).toMatchSnapshot()
})