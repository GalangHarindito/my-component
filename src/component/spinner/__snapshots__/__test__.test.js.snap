// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure input UI does not change unexpectedly 1`] = `
<div>
  <div
    class="loader loader-large"
  />
</div>
`;

exports[`Make sure input UI does not change unexpectedly 2`] = `
<div>
  <div
    class="loader loader-medium"
  />
</div>
`;

exports[`Make sure input UI does not change unexpectedly 3`] = `
<div>
  <div
    class="loader loader-small"
  />
</div>
`;
